src/librustc_mir/mir_map.rs

```diff
@@ -150,6 +143,32 @@ impl<'a, 'm, 'tcx> Visitor<'tcx> for InnerDump<'a,'m,'tcx> {
 
         intravisit::walk_fn(self, fk, decl, body, span);
     }
+
+    fn visit_item(&mut self, i: &'tcx hir::Item) {
+        match i.node {
+            hir::ItemConst(_, ref expr) |
+            hir::ItemStatic(_, _, ref expr) => {
+                let param_env = ty::ParameterEnvironment::for_item(self.tcx, i.id);
+                let infcx = infer::new_infer_ctxt(self.tcx,
+                                                  &self.tcx.tables,
+                                                  Some(param_env),
+                                                  ProjectionMode::AnyFinal);
+                let (mir, scope_auxiliary) = build::construct_expr(Cx::new(&infcx),
+                                                                   i.id,
+                                                                   i.span,
+                                                                   expr);
+                pretty::dump_mir(self.tcx,
+                                 "mir_map",
+                                 &0,
+                                 i.id,
+                                 &mir,
+                                 Some(&scope_auxiliary));
+                assert!(self.map.map.insert(i.id, mir).is_none());
```

Comment:
I guess it makes sense to generate a `const fn () -> T` for `const` and `static` initializers in order to be able to evaluate their value, but I would expect to see them stored as eagerly evaluated values, rather than keeping around representation of their initializers.

I realise we do not have a constant propagator for MIR yet, but a FIXME or a TODO in that direction would be nice.